core/Archive/ArchiveInvalidator.php

```diff
@@ -796,9 +796,11 @@ private function getEarliestDateToRearchive()
             return null;
         }
 
-        $lastNMonthsToInvalidate = (int) substr($lastNMonthsToInvalidate, 4);
-        if (empty($lastNMonthsToInvalidate)) {
-            return null;
+        if (!is_numeric($lastNMonthsToInvalidate)) {
+            $lastNMonthsToInvalidate = (int)str_replace('last', '', $lastNMonthsToInvalidate);
```

Comment:
In theory this would also allow values like  `last2last1` and would result in `21`. Not sure if we wouldn't want to discard such incorrect values instead.

Reply:
cc @tsteur

Reply:
I think it's not needed and could keep things simple. People will mostly just configure a number anyway.